apps/client/lib/services/auth.service.ts

```diff
@@ -19,7 +19,13 @@ export namespace AuthService {
 
   let authTokenListeners: AuthTokenCallback[] = [];
 
-  export function addAuthTokenListener(fn: AuthTokenCallback) {
+  export function addAuthTokenListener(
+    fn: AuthTokenCallback,
+    immediate: boolean = false,
+  ) {
+    if (immediate) {
+      fn(getAuthToken());
+    }
```

Comment:
The `immediate` flag allows subscribers to get the current state of the auth token as soon as they subscribe, which handles cases where it may have been set before they subscribed.